--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -116,7 +116,7 @@ configure_file(${CMAKE_CURRENT_SOURCE_DI
 )
 
 include(CMakePackageConfigHelpers)
-set(CONFIG_PACKAGE_LOCATION "${CMAKE_INSTALL_DATADIR}/${PROJECT_NAME}/cmake")
+set(CONFIG_PACKAGE_LOCATION "${CMAKE_INSTALL_DATADIR}/cmake/${PROJECT_NAME}")
 
 write_basic_package_version_file(
 	"${CMAKE_CURRENT_BINARY_DIR}/${EXPORT_TARGETS_NAME}ConfigVersion.cmake"
